docs/new-architecture-app-intro.md

```diff
@@ -0,0 +1,192 @@
+
+## Use a React Native nightly release
+
+At this time, you must use a React Native nightly release in order to get access to the most up to date changes. Eventually, we will recommend targeting a minimum stable open source release.
+
+This guide is written with the expectation that you’re using a specific nightly release. As new revisions of the playbook are release, the target nightly release may be updated. The specific nightly version that we will be using throughout the rest of the playbook is version `0.0.0-20211205-2008-583471bc4`.
+
+Before upgrading your app to a specific nightly release, we recommend upgrading your app to the latest open source release. By upgrading to a published open source release first, you will be able to take advantage of tools like the [upgrade helper](https://react-native-community.github.io/upgrade-helper/) to determine what other changes may be required for your project.
+
+As of this writing, the latest stable release is `0.66.3`. Once you have upgraded your project to this version successfully, you may proceed to targeting the `0.0.0-20211102-2008-8fef52035` nightly release. You may target this nightly release the same way you’d target any other version of React Native:
+
+```bash
+yarn add react-native@0.0.0-20211205-2008-583471bc4
+```
+
+## Install react-native-codegen v0.0.12 or above
+
+Make sure that you have the most recent react-native-codegen package; otherwise, you might see an error like `***TypeError: RNCodegen.generateFromSchemas is not a function.***`
+
+```bash
+yarn add react-native-codegen@0.0.12
+```
+
+If you are still getting the error, you may have an older version installed under node_modules/react-native/node_modules. You can remove that or reinstall everything in node_modules to fix the problem.
+
+### Android specifics
+
+Using the new architecture on Android has some prerequisites that you need to meet:
+
+1. Using Gradle 7.x and Android Gradle Plugin 7.x
+2. Using the **new React Gradle Plugin**
+3. Building `react-native` **from Source**
+
+You can update Gradle by running:
+
+```bash
+cd android && ./gradlew wrapper --gradle-version 7.3 --distribution-type=all
+```
+
+While the AGP version should be updated inside the **top level** `build.gradle` file at the `com.android.tools.build:gradle` dependency line.
+
+If you’re set with it, let’s now install the new Gradle plugin which is distributed through a NPM package called [**`react-native-gradle-plugin`**](https://www.npmjs.com/package/react-native-gradle-plugin). You can do so with:
+
+```bash
+yarn add react-native-gradle-plugin
+```
+
+You can control if you have the package already installed by doing:
+
+```bash
+ls -la node_modules/react-native-gradle-plugin
+```
+
+Now, you can edit your **top level** `settings.gradle` file to include the following line at the end of the file:
+
+```groovy
+includeBuild('../node_modules/react-native-gradle-plugin') {
```

Comment:
> I pulled branch`0.66-stable` and copied android folder

That's not the correct approach. There should not be an `android` folder inside `react-native-codegen` therefore this can't be the long term solution. I'm trying to see if I can reproduce this error locally, but it seems like this is not the case.